### PR TITLE
[CORE-67]: implement rich health check and fix bug with freezing account caching

### PR DESCRIPTION
**Pull request by Anglopol**

### CODE REVIEW

File: src/realms-repository.ts [hunk 1]
```diff
@@ -37,6 +43,11 @@ export interface RealmMints {
 
 @Injectable()
 export class RealmsRepository implements OnModuleInit {
+  private static readonly MAX_CACHING_EXECUTION_TIME_MILLS = process.env
+    .MAX_CACHING_EXECUTION_TIME_MILLS
```
Comment (tsmbl) [inline src/realms-repository.ts:47]: MILLIS

File: src/realms-repository.ts [hunk 2]
```diff
@@ -37,6 +43,11 @@ export interface RealmMints {
...
+    ? parseInt(process.env.MAX_CACHING_EXECUTION_TIME_MILLS, 10)
```
Comment (tsmbl) [inline src/realms-repository.ts:48]: Also MILLIS

File: src/caching.health.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    throw new HealthCheckError(
+      'Caching failed',
+      this.getStatus('caching', isHealthy),
+    );
+  }
+
+  @OnEvent(CachingEventType.Started)
+  incrementIngestionAttempts({ timeStarted, maxTimeout }: CachingStartedEvent) {
+    this.lastTimout = maxTimeout;
```
Comment (tsmbl) [inline src/caching.health.ts:49]: Time

File: src/realms-repository.ts [hunk 4]
```diff
@@ -66,10 +80,22 @@ export class RealmsRepository implements OnModuleInit {
       return;
     }
     this.cachingInProgress = true;
+    const cachingStartedEvent: CachingStartedEvent = {
+      timeStarted: Date.now(),
+      maxTimeout:
+        RealmsRepository.MAX_CACHING_EXECUTION_TIME_MILLS +
```
Comment (tsmbl) [inline src/realms-repository.ts:86]: Do we need to specify MAX_CACHING_EXECUTION_TIME_MILLS and SET_TIMEOUT_DELAY in event? These seem to be constants that can be just initialized inside indicator 
May be I'm wrong, but this adds extra state management but no benefits

File: src/caching.health.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      return this.getStatus('caching', isHealthy);
+    }
+    throw new HealthCheckError(
+      'Caching failed',
+      this.getStatus('caching', isHealthy),
+    );
+  }
+
+  @OnEvent(CachingEventType.Started)
+  incrementIngestionAttempts({ timeStarted, maxTimeout }: CachingStartedEvent) {
```
Comment (tsmbl) [inline src/caching.health.ts:48]: naming seems to be a bit confusing here

File: src/caching.health.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    this.lastTimout = maxTimeout;
+    this.lastStartedCaching = timeStarted;
+    this.cachingInProgress = true;
+  }
+
+  @OnEvent(CachingEventType.Finished)
+  resetIngestionAttempts() {
```
Comment (tsmbl) [inline src/caching.health.ts:55]: naming seems to be a bit confusing here

**Review by tsmbl — CHANGES_REQUESTED** [review]
Overall good solution and job, left several of comments

File: src/realms-repository.ts (hunk 4 above)
Comment (Anglopol) [reply src/realms-repository.ts:86]: Forgot to delete this after removing the timeout for caching account

**Review by tsmbl — APPROVED** [review]
lgtm, let's merge and deploy on Monday

**Pull request merged**